components/feature/contextmenu/src/main/java/mozilla/components/feature/contextmenu/ContextMenuCandidate.kt

```diff
@@ -159,6 +164,82 @@ data class ContextMenuCandidate(
+         */
+        fun createAddContactCandidate(
+            context: Context
+        ) = ContextMenuCandidate(
+            id = "mozac.feature.contextmenu.add_to_contact",
+            label = context.getString(R.string.mozac_feature_contextmenu_add_to_contact),
+            showFor = { _, hitResult -> hitResult.isMailto() },
+            action = { _, hitResult ->
+                val link = hitResult.getLink()
+                val intent = Intent(ContactsContract.Intents.Insert.ACTION).apply {
+                    type = ContactsContract.RawContacts.CONTENT_TYPE
+                    putExtra(ContactsContract.Intents.Insert.EMAIL, link.stripMailToProtocol())
+                    putExtra(
+                        ContactsContract.Intents.Insert.EMAIL_TYPE,
+                        ContactsContract.CommonDataKinds.Email.TYPE_WORK
+                    )
+                    addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
+                }
+
+                context.startActivity(intent)
+            }
+        )
+
+        /**
+         * Context Menu item: "Share email address".
+         */
+        fun createShareEmailAddressCandidate(
+            context: Context
+        ) = ContextMenuCandidate(
+            id = "mozac.feature.contextmenu.share_email",
+            label = context.getString(R.string.mozac_feature_contextmenu_share_email_address),
+            showFor = { _, hitResult -> hitResult.isMailto() },
+            action = { _, hitResult ->
+                val intent = Intent(Intent.ACTION_SEND).apply {
+                    type = "text/plain"
+                    flags = Intent.FLAG_ACTIVITY_NEW_TASK
+                    putExtra(Intent.EXTRA_TEXT, hitResult.getLink().stripMailToProtocol())
+                }
+
+                val shareIntent = Intent.createChooser(
+                    intent,
+                    context.getString(R.string.mozac_feature_contextmenu_share_email_address)
+                ).addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
+
+                context.startActivity(shareIntent)
+            }
+        )
+
+        /**
+         * Context Menu item: "Copy email address".
+         */
+        fun createCopyEmailAddressCandidate(
+            context: Context,
+            snackBarParentView: View,
+            snackbarDelegate: SnackbarDelegate = DefaultSnackbarDelegate()
+        ) = ContextMenuCandidate(
+            id = "mozac.feature.contextmenu.copy_email_address",
+            label = context.getString(R.string.mozac_feature_contextmenu_copy_email_address),
+            showFor = { _, hitResult -> hitResult.isMailto() },
+            action = { _, hitResult ->
+                val clipboardManager =
+                    context.getSystemService(Context.CLIPBOARD_SERVICE) as ClipboardManager
+                val email = hitResult.getLink().stripMailToProtocol()
+                val clip = ClipData.newPlainText(email, email)
+                clipboardManager.setPrimaryClip(clip)
```

Comment:
This could also we an extension function. What do yo think?

Reply:
Agreed.  I'll look into moving this.

Reply:
Couldn't find a good place to move this as extension.  But I've refactored it as a private function since there are 3 places that have similar usage.

Reply:
👍